Mathlib/RingTheory/Finiteness/Small.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir, María-Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, Maria-Inés de Frutos-Fernandez
+-/
+import Mathlib.LinearAlgebra.Finsupp.LinearCombination
+import Mathlib.RingTheory.FiniteType
+import Mathlib.LinearAlgebra.DFinsupp
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+import Mathlib.LinearAlgebra.Basis.Cardinality
+import Mathlib.LinearAlgebra.StdBasis
+import Mathlib.RingTheory.Finiteness.Basic
+
+/-! # Smallness properties of modules and algebras -/
+
+universe u
+
+namespace Submodule
+
+variable {R M : Type*} [Semiring R] [AddCommMonoid M] [Module R M]
+
+theorem bot_small : Small.{u} (⊥ : Submodule R M) := by
+  let f : Unit → (⊥ : Submodule R M) := fun _ ↦ 0
+  have : Small.{u} Unit := small_lift Unit
+  apply small_of_surjective (f := f)
+  rintro ⟨x, hx⟩
+  simp only [mem_bot] at hx
+  use default
+  simp [← Subtype.coe_inj, f, hx]
```

Comment:
Doesn't `inferInstance` find this? By going through [Countable.toSmall](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Countable/Small.html#Countable.toSmall) and [Subsingleton.to_countable](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Countable/Defs.html#Subsingleton.to_countable).